src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -89,9 +98,13 @@ class AppsResource @Inject() (
     async {
       implicit val identity = await(authenticatedAsync(req))
 
-      val rawApp = Raml.fromRaml(Json.parse(body).as[raml.App].normalize)
+      val ramlApp = Json.parse(body).as[raml.App]
+
+      implicit val (normalize, validate) = createValidatorAndNormalizerForApp(PathId(ramlApp.id))
+
       val now = clock.now()
-      val app = validateOrThrow(rawApp).copy(versionInfo = VersionInfo.OnlyVersion(now))
+
+      val app = validateOrThrow(Raml.fromRaml(ramlApp.normalize)).copy(versionInfo = VersionInfo.OnlyVersion(now))
```

Comment:
I wonder if it should just be `normalizer.normalized(ramlApp)`. This ways `normalize` does not have to be implicit.

Reply:
I don't really like the pattern for the Normalizer, but in this case i'd leave it as is, both validator and normalizer are implicit and it is kind of clean. Making only the normalizer explicit wouldn't really help much

Reply:
Fair point,